### ISSUE TITLE
Add good practices on `auto`

### ISSUE THREAD

**Issue by bernhardmgruber** [hsf-training/cpluspluscourse#423]
In the basic chapter we present the use of `auto` but we don't give enough guidance on when to use it. I think we should mention that you can also use too much `auto` and make the code hard to understand.

**Comment by hageboeck** [comment]
I agree with the comment, but in the past iterations, we always discussed this during the lecture. And it was fine to do that without extra slides on the topic.
I will close this now, but please reopen if you think that doing it orally isn't enough.
